include/aws/testing/aws_test_harness.h

```diff
@@ -35,6 +36,8 @@ the AWS_UNSTABLE_TESTING_API compiler flag
 #define AWS_TESTING_REPORT_FD stderr
 #endif
 
+static struct aws_mutex allocator_lock;
```

Comment:
Having static data members in a test header is pretty weird. Why isn't the malloc harness part of a library .c file?

Reply:
Having to link in a test lib will end up being a pain for users. Just look at google test and their move to amalgamated header file.

Plus it ends up being static to the main test entry point which I think is fine.

Reply:
It could be just part of the main aws-c-commons library.